jobs/jobs/utils.py

```diff
@@ -860,3 +896,20 @@ async def validate_create_job_previous_jobs(
             detail="Jobs with these ids do not exist.",
         )
     return [j.id for j in previous_jobs]
+
+
+async def update_create_job_params_using_revisions(
+    job_params: schemas.JobParams, current_tenant: str, jwt_token: str
+) -> None:
+    response = await get_annotations_by_revisions(
+        current_tenant=current_tenant,
+        jw_token=jwt_token,
+        revisions=list(job_params.revisions),
+    )
+
+    unique_file_ids_of_revisions = set(
+        [data["file_id"] for data in response.get("data", [])]
```

Comment:
what if payload changes? it will fail, let's add some processing here. at least check if file_id key is present, if not, skip